### PR TITLE
[4.0] Fixed the drag and drop installation failing due to missing data

### PR DESCRIPTION
**Pull request by roland-d**
Pull Request for Issue #22043 

### Summary of Changes
Added the data to be posted on installation

### Testing Instructions
1. Go to System -> Install -> Extension
2. Click on the Upload Package File tab
3. Drag and drop a package
4. Notice the installation fails
5. Apply patch
6. Run `npm install`
7. Go to System -> Install -> Extension
8. Click on the Upload Package File tab
9. Drag and drop a package
10. Notice the installation succeeds

### Expected result
Installation succeeds

### Actual result
Installation fails

### Documentation Changes Required
None

### CODE REVIEW

**Comment by renekreijveld** [comment]
I have tested this item :white_check_mark: successfully on 7e09a1138837ee259a42818fe618e11dd2cb7de4<br /><br />Tried installing a component.
When dragging and dropping a component the message "No Install Type Found" appears.
When choosing a file after browsing for file two messages appear:
- JInstaller: :Install: Can't find Joomla XML setup file. No data supplied for parameters in prepared statement
- Error installing component

Installed patch.
Installing through drag and drop and by browsing to file now works fine.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22071">issues.joomla.org/tracker/joomla-cms/22071</a>.</sub>

**Comment by crommie** [comment]
I have tested this item :white_check_mark: successfully on 7e09a1138837ee259a42818fe618e11dd2cb7de4<br /><br />Followed testing instructions. Outcome as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22071">issues.joomla.org/tracker/joomla-cms/22071</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

File: libraries/src/Installer/InstallerScript.php [hunk 1]
```diff
@@ -127,8 +127,9 @@ public function preflight($type, $parent)
 		}
 
 		// Extension manifest file version
-		$this->release = $parent->getManifest()->version;
-		$extensionType = substr($this->extension, 0, 3);
+		$this->extension = $parent->getName();
```
Comment (wilsonge) [inline libraries/src/Installer/InstallerScript.php:130]: Can you backport this part to staging as well please

**Pull request merged**